packages/shared/src/urbit/channel.ts

```diff
@@ -357,6 +371,53 @@ export interface Perm {
   group: Flag;
 }
 
+interface Bot {
+  nickname?: string;
+  avatar?: string;
+  origin?: string;
+  agent: string;
+}
+
+interface PostMetadataSchemaV1 {
+  type: string;
+  bot?: Bot;
+  rendererOverride?: string;
+}
+
+interface MessageInput {
+  type: string;
+  postType: string;
+  configuration: Record<string, string>;
+}
+
+interface ContentRenderer {
+  render: string;
+  using: string;
+}
+
+interface ChannelMetadataSchemaV1 {
+  version: 1;
+  messageInputs: MessageInput[];
+  defaultContentRenderers?: ContentRenderer[];
+  defaultPostCollectionRenderer?: string;
+}
+
+type ChannelMetadataSchema = ChannelMetadataSchemaV1;
+
+interface Hook {
+  version: number;
+  name: string;
```

Comment:
nit: it sounds like `name` needs to be unique – as a non-urbit user, `name` feels like "display name" instead of a machine id. `id` or `key` is more communicative of the required uniqueness

Reply:
yeah this is good feedback will change